src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs

```diff
@@ -228,21 +228,24 @@ private static void SerializeStateMachineLocalScopes(IMethodBody methodBody, Arr
             customDebugInfo.Add(cmw);
         }
 
-        private static void SerializeDynamicLocalInfo(IMethodBody methodBody, ArrayBuilder<PooledBlobBuilder> customDebugInfo)
+        private static bool ShouldSerializeLocalInfo(
+            IMethodBody methodBody,
+            ref ArrayBuilder<ILocalDefinition> builder,
```

Comment:
Do we need to pass builder by ref? It seems to be always null when called. Perhaps could be a return value for the method?